### PR TITLE
PSR-1564-Final | Implement optionality for remaining Loans fields

### PR DESCRIPTION
**Pull request by ccatosdevelopment**

### CODE REVIEW

File: test/resources/PSR with Login/Standard PSR -Pre-populated - Success.bru [hunk 1]
```diff
@@ -104,9 +99,7 @@ body:json {
               },
               "optConnectedPartyStatus": true,
               "totalCostOfLandOrProperty": 1000000,
-              "optIndepValuationSupport": true,
-              "optLandOrPropertyLeased": false,
-              "optTotalIncomeOrReceipts": 25000
```
Comment (ccatosdevelopment) [inline test/resources/PSR with Login/Standard PSR -Pre-populated - Success.bru:109]: Should have been removed in earlier ticket.

File: test/resources/PSR with Login/Standard PSR -Pre-populated - Success.bru [hunk 2]
```diff
@@ -133,9 +126,7 @@ body:json {
               },
               "optConnectedPartyStatus": true,
               "totalCostOfLandOrProperty": 1000000,
-              "optIndepValuationSupport": false,
-              "optLandOrPropertyLeased": false,
-              "optTotalIncomeOrReceipts": 25000
```
Comment (ccatosdevelopment) [inline test/resources/PSR with Login/Standard PSR -Pre-populated - Success.bru:138]: Unrelated to this ticket, should have been removed in earlier ticket.

File: test/resources/PSR with Login/Standard PSR -Pre-populated - Success.bru [hunk 3]
```diff
@@ -160,8 +151,7 @@ body:json {
               },
               "optConnectedPartyStatus": false,
               "totalCostOfLandOrProperty": 14000000,
-              "optIndepValuationSupport": false,
-              "optTotalIncomeOrReceipts": 500000
```
Comment (ccatosdevelopment) [inline test/resources/PSR with Login/Standard PSR -Pre-populated - Success.bru:164]: Unrelated to this ticket, should have been removed in earlier ticket.

File: test/resources/api-1999-land-or-property-only-valid-example-optional-fields.json [hunk 4]
```diff
@@ -31,7 +31,6 @@
   },
   "assets": {
     "landOrProperty": {
-      "noOfTransactions": 1,
```
Comment (ccatosdevelopment) [inline test/resources/api-1999-land-or-property-only-valid-example-optional-fields.json:34]: Unrelated to this ticket, should have been removed in earlier ticket.

File: app/uk/gov/hmrc/pensionschemereturn/transformations/Transformer.scala [hunk 5]
```diff
@@ -44,8 +44,6 @@ trait Transformer {
 
   protected def optToYesNo(optValue: Option[?]): String = optValue.map(_ => Yes).getOrElse(No)
 
-  protected def optToOptYesNo(optValue: Option[_]): Option[String] = optValue.map(_ => Some(Yes)).getOrElse(Some(No))
-
```
Comment (ccatosdevelopment) [inline app/uk/gov/hmrc/pensionschemereturn/transformations/Transformer.scala:48]: Unused.

**Review by cyeniceri — APPROVED** [review]

**Pull request merged**